src/main/java/com/biku89/medical_clinic/DoctorService.java

```diff
@@ -0,0 +1,52 @@
+package com.biku89.medical_clinic;
+
+import lombok.Data;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@RequiredArgsConstructor
+public class DoctorService {
+    private final DoctorRepository doctorRepository;
+    private final DoctorMapper doctorMapper;
+    private final InstitutionRepository institutionRepository;
+
+    public List<DoctorDTO> getAllDoctors(){
+        return doctorRepository.findAll().stream().map(doctorMapper::toDTO).toList();
+    }
+
+    public Optional<DoctorDTO> getDoctorByEmail(String email){
+        return doctorRepository.findByEmail(email).map(doctorMapper::toDTO);
```

Comment:
doctorRepository.findByEmail(email).map(doctorMapper::toDTO).orElseThrow () <- co ma sie stac jak nie ma takiego doktora to tutaj powinien byc powiedziane co ma sie stac a nie zrzucac odpowiedzialnosc na controller.